### ISSUE TITLE
broadcasting during evaluation?

### ISSUE THREAD

**Issue by bluesummers1129** [horovod/horovod#1208]
This might be a dumb question, but I'm a little confused about broadcasting. I see in the tensorflow_mnist_estimator example that as part of training we first broadcast the variables via the BroadcastGlobalVariables hook, but do not do this before evaluation:

```
    mnist_classifier.train(
        input_fn=train_input_fn,
        steps=20000 // hvd.size(),
        hooks=[logging_hook, bcast_hook])


    # Evaluate the model and print results
    eval_input_fn = tf.estimator.inputs.numpy_input_fn(
        x={"x": eval_data},
        y=eval_labels,
        num_epochs=1,
        shuffle=False)
    eval_results = mnist_classifier.evaluate(input_fn=eval_input_fn)
```
Why don't we have to send a broadcast hook during evaluation as well? I notice that when running the evaluate in a multi-worker setting I see messages like
```
[1,18]<stderr>:INFO:tensorflow:Restoring parameters from /tmp/tmphrhnffod/model.ckpt-10000
[1,20]<stderr>:INFO:tensorflow:Restoring parameters from /tmp/tmp70c22rr8/model.ckpt-10000
[1,20]<stderr>:INFO:tensorflow:Restoring parameters from /tmp/tmp70c22rr8/model.ckpt-10000
[1,0]<stderr>:INFO:tensorflow:Restoring parameters from /mnt/test_checkpoint_proximal3/model.ckpt-10000
[1,0]<stderr>:INFO:tensorflow:Restoring parameters from /mnt/test_checkpoint_proximal3/model.ckpt-10000
```
but shouldn't we want all the workers with hvd.rank() != 0 to also use the global model parameters?

**Comment by tgaddair** [comment]
Hey @bluesummers1129, the purpose of broadcasting is to ensure that all workers start from the same set of parameter values.  After we broadcast the initial parameters from rank 0 to the other workers, all workers should have the same set of parameters after training. That's because after we allreduce the gradients, all updates to model parameters should be the same.  In practice there is some numerical non-determinism that can cause them to diverge slightly, but it's usually minor enough to be ignored.  

As a result, there's no need to broadcast the model parameters because all workers have the same parameter values already.

**Comment by bluesummers1129** [comment]
That makes sense, thanks!
